### PR TITLE
Added PHP 7.0 compatibility

### PR DESCRIPTION
**Pull request by fred-jan**
Depends on:
- [x] https://github.com/MyOnlineStore/schema-registry-php-client/pull/1
- [x] https://github.com/MyOnlineStore/avro-php/pull/1
- [x] https://github.com/MyOnlineStore/php-functional/pull/1
- [x] https://github.com/MyOnlineStore/fantasy-land/pull/1

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,16 +22,35 @@
...
   "require": {
-    "php": "^7.1",
+    "php": "^7.0",
     "beberlei/assert": "~2.7",
-    "flix-tech/confluent-schema-registry-api": "^6.0.0",
-    "flix-tech/avro-php": "^3.0.0",
-    "widmogrod/php-functional": "^4.2|^5.0"
+    "flix-tech/confluent-schema-registry-api": "dev-backport-7.0",
+    "flix-tech/avro-php": "dev-backport-7.0",
+    "functional-php/fantasy-land": "dev-backport-7.0",
```
Comment (gertjuhh) [inline composer.json:48]: Why did you add this? It wasn't in the original list of requirements.

File: src/Objects/SchemaResolvers/CallableResolver.php [hunk 2]
```diff
@@ -30,7 +30,7 @@ public function valueSchemaFor($record): AvroSchema
         return AvroSchema::parse(\call_user_func($this->valueSchemaResolverCallable, $record));
     }
 
-    public function keySchemaFor($record): ?AvroSchema
+    public function keySchemaFor($record)
```
Comment (gertjuhh) [inline src/Objects/SchemaResolvers/CallableResolver.php:36]:
```suggestion
    /**
     * @return AvroSchema|null
     */
    public function keySchemaFor($record)
```

File: test/Objects/SchemaResolvers/DefinitionInterfaceResolverTest.php [hunk 3]
```diff
@@ -83,7 +83,7 @@ public static function valueSchemaJson(): string
                 return self::$valueSchema;
             }
 
-            public static function keySchemaJson(): ?string
+            public static function keySchemaJson()
```
Comment (gertjuhh) [inline test/Objects/SchemaResolvers/DefinitionInterfaceResolverTest.php:89]:
```suggestion
            /**
             * @return string|null
             */
            public static function keySchemaJson()
```

**Review by gertjuhh — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (fred-jan) [reply composer.json:48]: This was needed for Composer to be able to resolve the `dev-backport7.0` reference.

**Review by gertjuhh — APPROVED** [review]
